app/assets/javascripts/discourse/app/components/topic-list/topic-link.gjs

```diff
@@ -1,20 +1,34 @@
 import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import { service } from "@ember/service";
 import { htmlSafe } from "@ember/template";
 import PluginOutlet from "discourse/components/plugin-outlet";
 import lazyHash from "discourse/helpers/lazy-hash";
 
 export default class TopicLink extends Component {
+  @service prefetch;
+
   get url() {
     return this.args.topic.linked_post_number
       ? this.args.topic.urlForPostNumber(this.args.topic.linked_post_number)
       : this.args.topic.lastUnreadUrl;
   }
 
+  @action
+  triggerPrefetch() {
+    this.prefetch.register(
+      this.args.topic.id,
+      this.args.topic.last_read_post_number
+    );
+  }
```

Comment:
Ideally this would be made as a proper modifier. `@ember/render-modifiers` are a compatibility thing which are not recommended for general use.

Also if it was a proper modifier, you could pass the element reference straight through to the service (so it wouldn't have to do a potentially-mismatching `querySelector` lookup).  It could also have associated cleanup logic, so that the cache can be cleared when the link is removed from the DOM. That would avoid the need for the catch-all `clearPrefetchedTopics()` function